lib/queue.js

```diff
@@ -663,8 +612,8 @@ class Queue extends Emitter {
 
       // invariant: in this code path, this.running < this.concurrency, always
       // after spoolup, this.running + this.queued === this.concurrency
```

Comment:
Very much would like to replace this with a more maintainable pattern that specifically handles this concurrency management.